.github/workflows/sync_updates_to_oscal_content.yml

```
@@ -0,0 +1,174 @@
+          pip3 install gitpython PyYAML deepdiff Jinja2 xmldiff
+      # Step 4: Checkout the ComplianceAsCode/content-test-filtering repository
+      - name: Checkout (CTF)
+        uses: actions/checkout@v4  # Use the latest version of actions/checkout
+        with:
+          repository: huiwangredhat/content-test-filtering # TODO: updated to ComplianceAsCode/content-test-filtering
+          ref: filter_controls_update
+          path: ctf
+  
+      # Step 5: Detect content changes in the PR
+      - name: Detect content changes in the PR
+        run: |
+          # TODO:
+          # 1. replace the 13176 to $pr_number
+          # 2. replace remote_repo value to ${{ github.server_url }}/${{ github.repository }}
+          # 3. replace base value to master
+          python ctf/content_test_filtering.py pr \
+            --base d2d1faf6db138f649c235af0f701cdfc110a3ffb \
+            --remote_repo https://github.com/ComplianceAsCode/content \
+            --verbose --profile --rule --output json 13176 > output.json
+      - name: Test if there are no content changes
+        run: echo "CTF_OUTPUT_SIZE=$(stat --printf="%s" output.json)" >> $GITHUB_OUTPUT
+        id: ctf
+      - name: Print changes to content detected if any
+        if: ${{ steps.ctf.outputs.CTF_OUTPUT_SIZE != '0' }}
+        run: cat output.json
+
+      # Step 6: Setup the trestle-bot CLI run envrionment
+      - name: Checkout CaC repo
+        if: ${{ steps.ctf.outputs.CTF_OUTPUT_SIZE != '0' }}
+        uses: actions/checkout@v4
+        with:
+          repository: ComplianceAsCode/content
+          path: content
+      - name: Checkout OSCAL content repo
+        if: ${{ steps.ctf.outputs.CTF_OUTPUT_SIZE != '0' }}
+        uses: actions/checkout@v4
+        with:
+          repository: huiwangredhat/trestle-repo # TODO: need to update it to OSCAL content repo
+          path: oscal-content
+          token: ${{ secrets.OSCAL_ACCESS_PAT }} # TODO: The access token should be granted the content write permission
+      - name: Checkout trestle-bot repo
+        if: ${{ steps.ctf.outputs.CTF_OUTPUT_SIZE != '0' }}
+        uses: actions/checkout@v4
+        with:
+          repository: complytime/trestle-bot
+          path: trestle-bot
+      - name: Setup trestlebot
+        run: |
+          cd trestle-bot
+          python3.9 -m pip install --no-cache-dir --upgrade pip setuptools
+          python3.9 -m pip install --no-cache-dir "poetry==1.7.1"
+          poetry install
+
+      # Step 7: Sync CAC content updates to OSCAL content
+      - name: Sync cac content updates to oscal content
+        if: ${{ steps.ctf.outputs.CTF_OUTPUT_SIZE != '0' }}
+        run: |
+          cd trestle-bot
+          pr_number="${{ github.event.pull_request.number }}"
+          RHEL_PRODUCTS=('rhel8', 'rhel9', 'rhel10', 'ocp4', 'rhcos4', 'rhv4')
+          while IFS= read -r line; do
+            if echo "$line" | grep -q "controls"; then
```

```suggestion
            if [[ "$line" == *controls* ]]; then
```
This should be a little more efficient and simpler.